x-pack/plugins/enterprise_search/public/applications/index.test.ts

```diff
@@ -7,13 +7,19 @@
 import { coreMock } from 'src/core/public/mocks';
 import { renderApp } from '../applications';
 
+jest.mock('./shared/telemetry', () => ({
+  SendAppSearchTelemetry: jest.fn(() => null),
+}));
```

Comment:
Without this mock override, this test was failing/erroring on the new sendTelemetry helper (http.catch specifically - leading me to think http/context isn't being populated?)

This mock/listener fixes that, but I'm not super in love with that so def open to other thoughts!

Reply:
@constancecchen Yeah I'm not sure how deeply things are mocked in coreMock. We could alternatively do something like.

```
core.http.catch = jest.fn()
```

Reply:
To close the loop on the Slack convo - 9841ce4 fixes this. The issue was that `sendTelemetry` was written w/ promise chaining - therefore it wanted a mocked promise (to be able to .catch the returned promise), not just a `jest.fn()` (what the kibana's mocked http was providing).

Switch to async/await syntax and a generic try/catch block fixes the issue 🎉